zcncore/transactionauth.go

```diff
@@ -108,6 +112,14 @@ func (ta *TransactionWithAuth) sign(otherSig string) error {
 }
 
 func (ta *TransactionWithAuth) submitTxn() {
+	nonce := ta.t.txn.TransactionNonce
+	if nonce < 1 {
+		nonce = transaction.Cache.GetNextNonce(ta.t.txn.ClientID)
+	} else {
+		transaction.Cache.Set(ta.t.txn.ClientID, nonce)
+	}
+	ta.t.txn.TransactionNonce = nonce
```

Comment:
The t.submitTxn() below will force sign the transaction again, which will reset the signature added in line 135 below.